File: src/example_ntuple.cpp
#include <iostream>
#include <vector>

#include <TROOT.h>
#include <TH1F.h>
#include <TH1I.h>
#include <TCanvas.h>

#include "Ntuple.h"

int main(int argc, const char *argv[]) {
  gROOT->SetBatch(kTRUE);

  Ntuple n("DataTree");
  n.add("data.root");
  ULong64_t entries = n.GetEntries();
  std::cout << "Entries: " << entries << std::endl;

  TH1F gauss_h("gauss", "Gaussian distribution", 100, -5., 5.);
  TH1I uniform_h("uniform", "Uniform distribution", 101, 0, 101);
  for (ULong64_t entry = 0; entry < entries; ++entry) {
    n.GetEntry(entry);
    gauss_h.Fill(n.val<Double_t>("gauss_0"));
    uniform_h.Fill(n.val<Int_t>("uniform_0"));
  }

  TCanvas c("c", "c", 400, 400);
  gauss_h.Draw();
  c.SaveAs("gauss.ntuple_cpp.pdf");
  c.Clear();
  uniform_h.Draw();
  c.SaveAs("uniform.ntuple_cpp.pdf");

  return 0;
}

File: python/example_ntuple.py
#!/usr/bin/env python

import ROOT

from Ntuple import Ntuple

ROOT.gROOT.SetBatch(True)

n = Ntuple("DataTree")
n.add("data.root")
entries = n.GetEntries()
print "Entries: ", entries

gauss_h = ROOT.TH1F("gauss", "Gaussian distribution", 100, -5., 5.)
uniform_h = ROOT.TH1I("uniform", "Uniform distribution", 101, 0, 101)

for entry in range(entries):
    n.GetEntry(entry)
    gauss_h.Fill(n.val("gauss_0"))
    uniform_h.Fill(n.val("uniform_0"))

canvas = ROOT.TCanvas("c", "c", 400, 400)
gauss_h.Draw()
canvas.SaveAs("gauss.ntuple_py.pdf")
canvas.Clear()
uniform_h.Draw()
canvas.SaveAs("uniform.ntuple_py.pdf")
